BigTuna/View Controllers/Discover/CategoryHeader.swift

```diff
@@ -0,0 +1,199 @@
+    }()
+    
+    let categoryCollectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.alwaysBounceHorizontal = true
+        collectionView.alwaysBounceVertical = false
+        collectionView.backgroundColor = .white
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+    
+    func setupViews() {
+        addSubview(discoverLabel)
+        addSubview(categoryCollectionView)
+        
+        categoryCollectionView.dataSource = self
+        categoryCollectionView.delegate = self
+
+        categoryCollectionView.register(CategoryCell.self, forCellWithReuseIdentifier: categoryCellIdentifier)
+        
+        categoryCollectionView.topAnchor.constraint(equalTo: discoverLabel.bottomAnchor).isActive = true
+        
+        addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "H:|[v0]|", options: NSLayoutConstraint.FormatOptions(), metrics: nil, views: ["v0": categoryCollectionView]))
+        
+        addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "V:|[v0]-[v1]|", options: NSLayoutConstraint.FormatOptions(), metrics: nil, views: ["v0": discoverLabel, "v1": categoryCollectionView]))
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        return UIEdgeInsets(top: 0, left: 14, bottom: 0, right: 14)
+    }
+    
+    // Conforms to protocol UICollectionViewDelegate
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: categoryCellIdentifier, for: indexPath) as! CategoryCell
+        cell.updateConstraints()
+        cell.categoryLabel.text = data[indexPath.row]
+        cell.categoryImageView.image = UIImage(named: "SurprisedPikachu")
+        cell.categoryImageView.setRounded()
+        return cell
+    }
+
+    // Conforms to protocol UICollectionViewDelegate
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return data.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let width = collectionView.frame.width * widthMultiplier
+        let height = collectionView.frame.height * heightMultplier
+        return CGSize(width: width, height: height)
+    }
+    
+    func reloadHeader() {
+        // Based on https://stackoverflow.com/a/13611230
+        categoryCollectionView.reloadSections(NSIndexSet(index: 0) as IndexSet)
+    }
+
+}
+
+class CategoryCell: UICollectionViewCell {
```

Comment:
should prob be in it's own file

Reply:
Since CategoryCell is exclusively used by CategoryHeader we probably don't need to add an entirely new file. I can make this class fileprivate tho 😄😄